corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js

```diff
@@ -209,6 +210,25 @@ FormplayerFrontend.on("start", function (options) {
     }
 });
 
+FormplayerFrontend.reqres.setHandler('getCurrentAppId', function() {
+    // First attempt to grab app id from URL
+    var urlObject = Util.currentUrlToObject(),
+        user = FormplayerFrontend.request('currentUser'),
+        appId;
+
+    appId = urlObject.appId;
+
+    if (appId) {
+        return appId;
+    }
+
+    // If it's not in the URL, then we are either on the home screen of formplayer
+    // and there is no app selected, or we are in preview mode.
+    appId = user.previewAppId;
```

Comment:
what do you mean by set? it uses it in the route but doesn't save it anywhere. the id exists in the # url fragment. we could add it to the url in the url object util. is that what you mean? that might be nicer than attaching it the user